Fix tutorial using abstract History type (fixes #150)

ALSchwalm opened the pull request:
I went ahead and just removed the explicit usage of History. I'm not sure if you'd prefer to use FileHistory to show that off. If so, I can change it to use that.

jonathanslenders commented:
Hi @ALSchwalm, thanks for the pull request!
I guess it's better to create an ``InMemoryHistory`` instance instead. The reason is that otherwise, for each call of ``get_input``, a new ``InMemoryHistory`` instance is created, and basically that way, there is no history between consecutive calls of ``get_input``.

ALSchwalm commented:
Yep, I should have tested this a bit more thoroughly. Fixed and rebased. Thanks for your work on this project, by the way. I'm working on some stuff with it and It's been a pleasure to use.

jonathanslenders commented:
Thanks you! Merged in: https://github.com/jonathanslenders/python-prompt-toolkit/commit/c1170c9b72eec9f1b57fd96a515dc2609bacfda6

(I also appreciate the rebase ;) )
